web/concrete/src/Service/Configuration/HTTP/ApacheGenerator.php

```diff
@@ -0,0 +1,110 @@
+<?php
+namespace Concrete\Core\Service\Configuration\HTTP;
+
+use Concrete\Core\Service\Configuration\GeneratorInterface;
+use Concrete\Core\Application\Application;
+
+class ApacheGenerator implements GeneratorInterface
+{
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var array
+     */
+    protected $rules;
+
+    /**
+     * @var array
+     */
+    protected $enabledRules;
+
+    /**
+     * Initializes the instance.
+     *
+     * @param Application $app
+     */
+    public function __construct(Application $app)
+    {
+        $DIR_REL = DIR_REL;
+        $DISPATCHER_FILENAME = DISPATCHER_FILENAME;
+        $this->app = $app;
+        $this->rules = array();
+        $this->enabledRules = array();
+        $this->addRule(
```

Comment:
This spacing makes reading really hard, what with the multiple levels of indendation and here-doc, breaking this array out into its own method would be good since it'd be easier to read and slightly easier to modify:

```php
$this->addRule('pretty_urls', $this->getPrettyUrlRule(), true);
```